source/OptimaJet.DWKit.StarterApplication/Services/HttpOrganizationContext.cs

```diff
@@ -0,0 +1,51 @@
+﻿using System;
+using Microsoft.AspNetCore.Http;
+
+
+namespace OptimaJet.DWKit.StarterApplication.Services
+{
+    public class HttpOrganizationContext : IOrganizationContext
+    {
+        public static string ORGANIZATION_HEADER = "Organization";
+
+        public HttpContext HttpContext { get; set; }
+        public OrganizationService OrganizationService { get; set; }
+
+        public HttpOrganizationContext 
+        (
+            IHttpContextAccessor httpContextAccessor,
+            OrganizationService organizationService
+        )
+        {
+            this.HttpContext = httpContextAccessor.HttpContext;
+            this.OrganizationService = organizationService;
+            Update();
```

Comment:
rather than calling update in the constructor, the methods / properties should be memoized.

(having side-effects in constructors makes testing more difficult)

Reply:
Explain how it should be different?

Reply:
for example, for the CurrentUser,
![image](https://user-images.githubusercontent.com/199018/43227033-8270f8dc-902b-11e8-9c0e-fef42130b038.png)

tihs defines a getter property that is memoized such that subsequent calls don't make db hits again.

the CurrentUser is stashed in the HttpContext, but could probably just be a private property on that class since scoped services last the lifetime of a request.